src/components/listingPage/listingPage.component.js

```diff
@@ -162,6 +162,25 @@ class ListingPage extends Component {
               (res) => (this.state.listingRatings[i].userName = res.data.name)
             );
         }
+        //Determine if user has reserved listing
+        app
+          .get("/reservation/getByUser", {
+            headers: {
+              Authorization: `Bearer ${this.props.userSession.token}`,
+            },
+          })
+          .then(
+            (res) => {
+              res.data.reservations.forEach((reservation) => {
+                if (reservation._id === this.props.match.params.id) {
```

Comment:
this should be `reservation.listing` instead of `reservation._id` since we want to compare the reservation's listingId with the listingId in the url